RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc

```diff
@@ -195,10 +228,29 @@ class PFRecoTauDiscriminationByIsolation : public PFTauDiscriminationProducerBas
   double offsetRelativeSumPt_;
   double customIsoCone_;
   
+  bool applyPhotonPtSumOutsideSignalConeCut_;
+  double maxAbsPhotonSumPt_outsideSignalCone_;
+  double maxRelPhotonSumPt_outsideSignalCone_;
+
+  bool applyFootprintCorrection_;
+  struct footprintCorrectionType
+  {
+    footprintCorrectionType(const std::string& selection, const std::string& offset)
+      : selection_(selection),
+	offset_(offset)
+    {}
+    ~footprintCorrectionType() {}
+    StringCutObjectSelector<PFTau> selection_;
+    StringObjectFunction<PFTau> offset_;
+  };
+  std::vector<footprintCorrectionType*> footprintCorrections_;
```

Comment:
Ok, I changed every instance of footprintCorrectionType* to std::unique_ptr<FootprintCorrection> and am now getting the following compiler error:

/afs/cern.ch/work/a/ajohnson/CMSSW_7_6_X_2015-08-04-2300/src/RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc:117:102: error: conversion from 'PFRecoTauDiscriminationByIsolation::FootprintCorrection*' to non-scalar type 'std::unique_ptr<PFRecoTauDiscriminationByIsolation::FootprintCorrection>' requested
  std::unique_ptr<FootprintCorrection> footprintCorrection = new FootprintCorrection(selection, offset);